tests/components/laundrify/test_config_flow.py

```diff
@@ -0,0 +1,193 @@
+            context={CONF_SOURCE: SOURCE_USER},
+            data={CONF_CODE: "invalidFormat"},
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_CODE: "invalid_format"}
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    with _patch_laundrify_exchange_code() as laundrify_mock:
+        laundrify_mock.side_effect = exceptions.UnknownAuthCode
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_USER},
+            data=VALID_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_CODE: "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant):
+    """Test we handle cannot connect error."""
+    with _patch_laundrify_exchange_code() as laundrify_mock:
+        laundrify_mock.side_effect = exceptions.ApiConnectionException
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_USER},
+            data=VALID_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_unkown_exception(hass: HomeAssistant):
+    """Test we handle all other errors."""
+    with _patch_laundrify_exchange_code() as laundrify_mock:
+        laundrify_mock.side_effect = Exception
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_USER},
+            data=VALID_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_step_reauth(hass: HomeAssistant) -> None:
+    """Test the reauth form is shown."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_REAUTH}
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_FORM
+
+
+async def test_integration_already_exists(hass: HomeAssistant):
+    """Test we only allow a single config flow."""
+    create_entry(hass)
+    with _patch_laundrify_exchange_code(), _patch_laundrify_get_account_id():
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+        )
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_CODE: VALID_AUTH_CODE,
+            },
+        )
+
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_setup_entry_api_unauthorized(hass: HomeAssistant):
```

Comment:
the tests from here on aren't related to the config flow. They should be put in `test_init` instead